src/components/organisms/newProjectModal.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useMemo, useState } from "react";
+
+import { useForm } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+import { useNavigate } from "react-router-dom";
+
+import { ModalName, SidebarHrefMenu } from "@enums/components";
+import { defaultProjectFile } from "@src/constants";
+import { useModalStore, useProjectStore, useToastStore } from "@src/store";
+
+import { Button, ErrorMessage, Input, Loader } from "@components/atoms";
+import { Modal } from "@components/molecules";
+
+export const NewProjectModal = () => {
+	const { t } = useTranslation("modals", { keyPrefix: "newProject" });
+	const [isCreating, setIsCreating] = useState(false);
+	const { closeModal } = useModalStore();
+	const { projectsList } = useProjectStore();
+	const addToast = useToastStore((state) => state.addToast);
+	const { createProject } = useProjectStore();
+	const projectNamesSet = useMemo(() => new Set(projectsList.map((project) => project.name)), [projectsList]);
+	const navigate = useNavigate();
+
+	const {
+		formState: { errors },
+		handleSubmit,
+		register,
+		setValue,
+	} = useForm<{ projectName: string }>({
+		mode: "onChange",
+		defaultValues: {
+			projectName: "",
+		},
+	});
+
+	const validateProjectName = (value: string) => {
+		if (projectNamesSet.has(value)) {
+			return t("nameTaken");
+		}
+		if (!new RegExp("^[a-zA-Z_][\\w]*$").test(value)) {
+			return t("invalidName");
+		}
+	};
+
+	const onSubmit = async (data: { projectName: string }) => {
+		const { projectName } = data;
+
+		setIsCreating(true);
+		const { data: newProjectResponse, error } = await createProject(projectName, true);
+		setIsCreating(false);
+
+		if (error) {
+			addToast({
+				message: t("errorCreatingProject"),
+				type: "error",
+			});
+
+			return;
```

Comment:
add an indication of error inside the modal as well.
In this modal and in the templates too.

Reply:
done